dao-contracts/src/onboarding_voter.rs

```diff
@@ -0,0 +1,213 @@
+            fn get_dust_amount(&self) -> U256;
+            fn get_voting(&self, voting_id: U256) -> Option<Voting>;
+            fn get_ballot(&self, voting_id: U256, address: Address) -> Option<Ballot>;
+            fn get_voter(&self, voting_id: U256, at: u32) -> Option<Address>;
+        }
+    }
+
+    fn create_voting(&mut self, action: onboarding::Action, subject_address: Address, stake: U256) {
+        self.assert_no_ongoing_voting(&subject_address);
+
+        let (entry_point, runtime_args) = match action {
+            onboarding::Action::Add => self.configure_add_voting(subject_address),
+            onboarding::Action::Remove => self.configure_remove_voting(subject_address),
+        };
+        let creator = caller();
+        let contract_to_call = self.get_va_token_address();
+
+        self.voting
+            .create_voting(creator, stake, contract_to_call, entry_point, runtime_args);
+        self.onboarding.set_voting(&subject_address);
+    }
+
+    fn vote(&mut self, voting_id: VotingId, choice: Choice, stake: U256) {
+        let voter = caller();
+        self.voting.vote(voter, voting_id, choice, stake);
+    }
+
+    fn finish_voting(&mut self, voting_id: VotingId) {
+        let address = self.extract_address_from_args(voting_id);
+        self.voting.finish_voting(voting_id);
+        self.onboarding.clear_voting(&address);
+    }
+}
+
+// non-contract implementation
+impl OnboardingVoterContract {
+    fn configure_remove_voting(&mut self, subject_address: Address) -> (String, RuntimeArgs) {
+        self.assert_onboarded(&subject_address);
+        self.assert_has_reputation(&subject_address);
+
+        let token_id = self.onboarding.token_id_of(&subject_address).unwrap();
+
+        let runtime_args = runtime_args! {
+            ARG_TOKEN_ID => token_id,
+        };
+        let entry_point = ENTRY_POINT_BURN.to_string();
+
+        (entry_point, runtime_args)
+    }
+
+    fn configure_add_voting(&mut self, subject_address: Address) -> (String, RuntimeArgs) {
+        self.assert_not_onboarded(&subject_address);
+        self.assert_kyced(&subject_address);
+        self.assert_has_reputation(&subject_address);
+
+        let token_id = self.sequence.next_value();
+
+        let runtime_args = runtime_args! {
+            ARG_TO => subject_address,
+            ARG_TOKEN_ID => token_id,
+        };
+        let entry_point = ENTRY_POINT_MINT.to_string();
+
+        (entry_point, runtime_args)
+    }
+
+    fn extract_address_from_args(&self, voting_id: VotingId) -> Address {
+        let voting = self
+            .voting
+            .get_voting(voting_id)
+            .unwrap_or_revert_with(Error::VoterDoesNotExist);
+
+        let arg = voting
+            .runtime_args()
+            .named_args()
+            .find(|arg| arg.name() == ARG_TO);
+
+        if let Some(to_arg) = arg {
+            return to_arg.cl_value().clone().into_t().unwrap();
```

Comment:
unwrap

Reply:
Yup, but here we know the type, we know the operation is rather safe, so imo `unwrap_or_revert()` would be too much.